### PR TITLE
re-create 141

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
@gianarb this is for your wrong rebase :p

File: src/App.php [hunk 1]
```diff
@@ -133,15 +133,13 @@ public function run($request = null, $response = null)
...
      * Handle Route.
      *
-     * @param RouteInfoInterface $routeInfo
+     * @param mixed $routeInfo
      * @param EventInterface $event
      *
      * @throws RuntimeException if dispatch does not return RouteInfo object.
      */
-    private function handleRoute(
-        RouteInfoInterface $routeInfo,
```
Comment (samsonasik) [inline src/App.php:142]: if the Type hint is RouteInfoInterface, the RuntimeException won't be executed.

File: src/App.php [hunk 2]
```diff
@@ -133,15 +133,13 @@ public function run($request = null, $response = null)
...
-        EventInterface $event
-    ) {
+    private function handleRoute($routeInfo, EventInterface $event)
+    {
         if (!$routeInfo instanceof RouteInfoInterface) {
```
Comment (samsonasik) [inline src/App.php:143]: this check won't work if the Type hint is RouteInfoInterface, that's why `@param mixed` is passed. /cc @gianarb @EmanueleMinotto

File: src/App.php (hunk 1 above)
Comment (gianarb) [inline src/App.php:142]: mm ok thanks :)sorry

**Pull request merged**